### PR TITLE
Print unknown arguments when executing GraphStorm command-lines.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
#799 

*Description of changes:*
Add Warnings for unknown arguments. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by thvasilo** [comment]
Do we want warnings or error out when unknown args are provided?

I would vote for error out (which is also the default Python behavior), to ensure we honor the user's intent.

e.g. if the user provides:

```
--use-node-embedings true # this is a typo
```

the default value (false) will be used instead, and the only way for the user to know something went wrong is if they dig into the logs, which are hard to parse with all the printouts.

**Comment by classicsong** [comment]
> Do we want warnings or error out when unknown args are provided?
> 
> I would vote for error out (which is also the default Python behavior), to ensure we honor the user's intent.
> 
> e.g. if the user provides:
> 
> ```
> --use-node-embedings true # this is a typo
> ```
> 
> the default value (false) will be used instead, and the only way for the user to know something went wrong is if they dig into the logs, which are hard to parse with all the printouts.

There are two concerns for error out:
1) We have to add all the arguments from remap_result.py into the training loop, which is meaningless.
2) Originally, people can customize the GSConfig by just passing those "unknown arguments" and they can access them through config.udf_1, config.udf_2. If we error out, we will lose this flexibility.

**Review by thvasilo — APPROVED** [review]
Approve but let's keep in mind that if we run into problems in the future we can re-visit this decision.
